### PR TITLE
Restore release/0.5 JSON decode semantics and API signatures

### PR DESCRIPTION
**Pull request by hhenson**
## Summary

Restore two release/0.5 JSON decode behaviours and make the generated/runtime Python call signatures match the released public operators exactly.

- A bare TSS array is an additive tick delta; members absent from the payload remain present. Explicit `{"added": [...], "removed": [...]}` remains the removal form.
- A `null` element in a TSL array means that element does not tick.
- The public call shapes are `to_json(ts, delta=False)` and `from_json(ts)`.

## Correction to the original diagnosis

The first version of this PR confused internal implementation overloads with the public Python API. In hgraph 0.5.42, `from_json_generic(ts, _tp=..., delta=False)` is an internal implementation signature. The exported operators reject `_tp`, and exported `from_json` also rejects `delta`.

The compatibility shim, public `from_json` scalar, and generated `_tp` typing added on that premise have therefore been removed. Regression tests now assert both the exact public signatures and rejection of those non-public arguments.

## Implementation

- Decode bare TSS arrays through a set mutation, adding each parsed element without replacing current membership.
- Decode TSL arrays element-by-element so `null` can leave one element unticked.
- Give both native `to_json` overloads the same `delta=false` default so their public surface deduplicates correctly.
- Curate the two public runtime signatures to match 0.5.42 exactly and let `eval_node` fall back to the native operator contract when a curated signature intentionally has no annotations.
- Regenerate the operator catalogue, API inventory, docstrings, and typing protocol.

## Why the audit missed this

The existing generated API inventory describes the candidate native registry; it is not a comparison against the released Python package. Functional parity coverage also had `to_json(delta=...)` but no multi-tick `from_json` cases for TSS or partial TSL updates. That let both missing functionality and an incorrect interpretation of an internal signature pass as parity.

A separate follow-up will strengthen the differential parity surface and add a JSON behaviour matrix rather than widening this corrective PR.

## Verification

- macOS fresh native build: **1,655/1,655 passed**
- macOS stable-ABI wheel built with Python 3.12, installed and tested with Python 3.14: **2,219 passed, 10 skipped**
- private Linux fresh native build: **1,655/1,655 passed**
- private Linux stable-ABI wheel built with Python 3.12, installed and tested with Python 3.14: **2,219 passed, 10 skipped**
- focused JSON Python suite: **59 passed**
- API inventory, native docstring, and signature-compatibility tests: **29 passed**

## Compatibility note

This intentionally changes 0.8.20's handling of bare TSS arrays: they no longer remove members omitted from a later payload. That restores the released 0.5 delta semantics and avoids silent membership loss.

### CODE REVIEW

**Comment by hhenson** [comment]
Both Codex findings addressed in `0297a0fc8`.

**P2 — positional `_tp` carrier.** Confirmed against a real `hgraph==0.5.42` install rather than taken on inspection, and the finding is correct:

| call | 0.5.42 | #625 before | #625 after |
|---|---|---|---|
| `from_json[TS[int]](ts, TS[int])` | OK | **WiringError** | OK |
| `to_json(ts, TS[int], True)` | OK | **WiringError** | OK |

The guard only inspected `kwargs`, so the type expression landed on the new native `delta` scalar. The carrier is now recognised positionally too, and dropping it shifts 0.5's third argument onto `delta`. The ported suite gains these shapes including the three-argument `to_json` spelling.

**P2 — typing surface.** Also correct: the runtime accepted `_tp` while the generated protocol admitted only `ts` and `delta`, so a type checker would reject a working call. `tools/api_inventory.py` now carries a table of release/0.5 keywords absorbed at the Python boundary and appends them when rendering, so both operators get `_tp: _Any = ...`:

```
def __call__(self, ts: _WiringPort | str, delta: bool = ..., _tp: _Any = ...) -> _WiringPort: ...
```

Verification unchanged: parity sweep against 0.5.42 still **0 divergences of 7**; native **1655/1655**, python **2272 passed**, persistence **82**, fabric **30**.

Two unrelated problems found while validating this PR are filed separately rather than folded in: #626 (extension test suites cannot be collected together — duplicate module basenames) and #627 (segfault in `_hgraph` when persistence, fabric and analytics run in one process, which #626 had been masking).

File: python/hgraph/_wiring/_core.py [hunk 1]
```diff
@@ -596,6 +597,32 @@ def _normalise_type_arguments(self, args, kwargs):
...
+        information — ``to_json`` resolves the type from its input and
+        ``from_json`` from the requested output — so ``_tp`` is redundant rather
+        than missing, and is absorbed here instead of being reintroduced as a
+        native scalar.
+
+        ``delta`` is NOT handled here: it is a real native scalar on both
+        operators, so it passes straight through to dispatch.
+        """
+        if self.__name__ not in ("to_json", "from_json") or "_tp" not in kwargs:
+            return args, kwargs
```
[the remark this answers is not in the document]
Comment (hhenson) [reply python/hgraph/_wiring/_core.py]: Rechecked against the exported operators in a real 0.5.42 install, and the premise behind this finding was wrong: `_tp` is positional only on the internal generic implementation, while public `to_json` and `from_json` both reject it. Commit `c90b9a258` removes the positional/keyword shim and adds tests for the exact public signatures and rejection of `_tp`.

File: python/hgraph/_operator_typing.pyi [hunk 2]
```diff
@@ -2249,14 +2254,14 @@ class _from_json_Operator(_Protocol):
...
-    - ``from_json(ts: TS[str]) -> OUT``
+    - ``from_json(ts: TS[str], delta: bool = ...) -> OUT``
 
     Time-series parameters accept wiring ports and compatible plain
     values that can be lifted to constant sources. Generic names use
     the public Python vocabulary: ``SCALAR``, ``TIME_SERIES_TYPE``,
     ``SIZE``, ``OUT``, ``K`` and ``V``."""
 
-    def __call__(self, ts: _WiringPort | str) -> _WiringPort: ...
+    def __call__(self, ts: _WiringPort | str, delta: bool = ...) -> _WiringPort: ...
```
[the remark this answers is not in the document]
Comment (hhenson) [reply python/hgraph/_operator_typing.pyi]: This followed from the same mistaken premise. Since `_tp` is not accepted by the released public operators, the typing protocol must not advertise it. Commit `c90b9a258` removes that generated parameter; the public protocols now describe `to_json(ts, delta=False)` and `from_json(ts)`.

**Comment by hhenson** [comment]
Correction to my earlier review reply: it was inaccurate. I had verified the signature and behaviour of the internal `to_json_generic` / `from_json_generic` implementation functions and treated those as the public operator API. A direct check of the exported operators in hgraph 0.5.42 shows the public contracts are `to_json(ts, delta=False)` and `from_json(ts)`; `_tp` is rejected on both, and `delta` is rejected by `from_json`.

`c90b9a258` removes the resulting compatibility shim, the incorrect native `from_json` scalar, and the `_tp` typing additions. It also adds exact runtime-signature tests and negative tests for the non-public arguments. The valid TSS additive-delta and TSL-null decode fixes remain. The PR description and validation counts have been corrected accordingly.

**Comment by hhenson** [comment]
The differential parity follow-up is #628. It adds release/0.5 comparisons for the exact public `to_json` / `from_json` call shapes and their multi-tick JSON behavior, and fixes broad surface discovery so one failing lazy optional export is recorded rather than aborting the audit. This directly covers the audit gap that allowed the earlier `_tp` / `from_json(delta=...)` misdiagnosis.
